### ISSUE TITLE
False positive no-value-for-parameter when unpacking a modified list

### ISSUE THREAD

**Issue by IRDonch** [pylint-dev/pylint#4554]
### Steps to reproduce

Create `bug.py`:

```python
import os.path

def bad():
    a = []
    a.extend(['a', 'b'])

    return os.path.join(*a)
```

Run `pylint bug.py`.

### Current behavior

Pylint prints:

```
bug.py:7:11: E1120: No value for argument 'a' in function call (no-value-for-parameter)
```

### Expected behavior

No errors should be produced.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.8.3
astroid 2.5.6
Python 3.6.9 (default, Jan 26 2021, 15:33:00)
[GCC 8.4.0]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I can reproduce this.

**Comment by jontwo** [comment]
Is this likely to be fixed? Still present in 2.13.7.

**Comment by Pierre-Sassoulas** [comment]
There's a lot of open issues in pylint, so most of them are fixed when:
- Someone external find an issue annoying and fix it themself
- There's a lot of thumbs-up on the issue and a maintainer prioritize the issue

I think the best chance for this one to be fixed if is an external contributor propose a merge request.
